dcicutils/license_utils.py

```diff
@@ -757,141 +1168,242 @@ class C4InfrastructureLicenseChecker(LicenseChecker):
+                # Ref: https://www.npmjs.com/package/node-forge?activeTab=code
+                'node-forge',
+            ],
+
+            'MIT*': [
+
+                # This library uses a mix of licenses, but they (MIT, CC0) generally seem permissive.
+                # (It also mentions that some tools for building/testing use other libraries.)
+                # Ref: https://github.com/requirejs/domReady/blob/master/LICENSE
+                'domready',
+
+                # This library is under 'COMMON DEVELOPMENT AND DISTRIBUTION LICENSE (CDDL) Version 1.1'
+                # Ref: https://github.com/javaee/jsonp/blob/master/LICENSE.txt
+                # About CDDL ...
+                # Linking = Permissive, Private Use = ?
+                # Ref: https://en.wikipedia.org/wiki/Comparison_of_free_and_open-source_software_licenses
+                'jsonp',
+
+                # This library says pretty clearly it intends MIT license.
+                # Ref: https://www.npmjs.com/package/component-indexof
+                # Linking = Permissive, Private Use = Yes
+                # Ref: https://en.wikipedia.org/wiki/Comparison_of_free_and_open-source_software_licenses
+                'component-indexof',
+
+                # These look like a pretty straight MIT license.
+                # Linking = Permissive, Private Use = Yes
+                # Ref: https://en.wikipedia.org/wiki/Comparison_of_free_and_open-source_software_licenses
+                'mixin',        # LICENSE file at https://www.npmjs.com/package/mixin?activeTab=code
+                'stack-trace',  # https://github.com/stacktracejs/stacktrace.js/blob/master/LICENSE
+                'typed-function',  # LICENSE at https://www.npmjs.com/package/typed-function?activeTab=code
+
+            ],
+
+            'UNLICENSED': [
+                # The udn-browser library is our own and has been observed to sometimes show up in some contexts
+                # as UNLICENSED, when really it's MIT.
+                # Ref: https://github.com/dbmi-bgm/udn-browser/blob/main/LICENSE
+                'udn-browser',
+            ],
+        })
+
+
+@LicenseCheckerRegistry.register_checker('c4-infrastructure')
+class C4InfrastructureLicenseChecker(ParkLabCommonServerLicenseChecker):
+    """
+    Checker for C4 infrastructure (Fourfront, CGAP, SMaHT) from Park Lab.
+    """
 
-    }
+    LICENSE_TITLE = "(The )?MIT License"
 
 
+@LicenseCheckerRegistry.register_checker('c4-python-infrastructure')
 class C4PythonInfrastructureLicenseChecker(C4InfrastructureLicenseChecker):
     """
-    For situations like dcicutils and dcicsnovault where there's no Javascript, this will test just Python.
+    Checker for C4 python library infrastructure (Fourfront, CGAP, SMaHT) from Park Lab.
     """
     LICENSE_FRAMEWORKS = ['python']
+
+
+@LicenseCheckerRegistry.register_checker('scan2-pipeline')
+class Scan2PipelineLicenseChecker(ParkLabGplPipelineLicenseChecker):
```

Comment:
Could you explain why this needs a separate class? Is this because it uses mixed software not living under one language?

Reply:
We only have this one worked case of someone new working on things, but my definite sense from the experience is that this is one of those places where exceptions are the most likely. Every now and then you can make exceptions on a broad basis, like for docutils, because it's simply the nature of docutils that people mostly use those at dev time. But for other libraries, the thing is that you only know you can make an exception if the use is conforming because of how that particular repo uses something, like we know we only use it in some maintenance script and it's not part of the core library that gets delivered to pypi, or something like that. So I can't put that kind of exception in the main classes. But there could be some other place these are kept. Maybe someplace maintained by an employer or a corporate division (in our case dbmi or hms).